File: Samples/Wcf/NetCoreService/Startup.fs
namespace Softellect.Samples.Wcf.NetCoreService

open CoreWCF
open CoreWCF.Configuration
open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Hosting
open Microsoft.Extensions.DependencyInjection

open Softellect.Samples.Wcf.NetCoreService.EchoService

module Startup =

    type Startup() =

        let createServiceModel (builder : IServiceBuilder) =
            builder
                .AddService<EchoService>()
                .AddServiceEndpoint<EchoService, IEchoService>(BasicHttpBinding(), "/basichttp")
                .AddServiceEndpoint<EchoService, IEchoService>(NetTcpBinding(), "/nettcp")
            |> ignore

        member _.ConfigureServices(services : IServiceCollection) =
            services.AddServiceModelServices() |> ignore

        member _.Configure(app : IApplicationBuilder, _ : IWebHostEnvironment) =
            app.UseServiceModel(fun builder -> createServiceModel builder) |> ignore

File: WebAppMvc/Startup.cs
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.AspNetCore.HttpsPolicy;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Runtime.Versioning;
using System.Threading.Tasks;

namespace WebAppMvc
{
    public class Startup
    {
        /// <summary>
        /// https://weblog.west-wind.com/posts/2018/Apr/12/Getting-the-NET-Core-Runtime-Version-in-a-Running-Application
        /// </summary>
        public static string GetNetVersion()
        {
            var framework =
                //Environment.Version.ToString();

                System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription;

                //Assembly
                //    .GetEntryAssembly()?
                //    .GetCustomAttribute<TargetFrameworkAttribute>()?
                //    .FrameworkName;

            return framework;
        }

        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddControllersWithViews();
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }
            else
            {
                app.UseExceptionHandler("/Home/Error");
                // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
                app.UseHsts();
            }
            app.UseHttpsRedirection();
            app.UseStaticFiles();

            app.UseRouting();

            app.UseAuthorization();

            app.UseEndpoints(endpoints =>
            {
                endpoints.MapControllerRoute(
                    name: "default",
                    pattern: "{controller=Home}/{action=Index}/{id?}");
            });
        }
    }
}
